scipy/stats/tests/test_stats.py

```diff
@@ -538,6 +538,98 @@ def test_gh3014(self):
         odds, pvalue = stats.fisher_exact([[1, 2], [9, 84419233]])
 
 
+class TestBarnardExact(object):
+    """Some tests to show that barnard_exact() works correctly.
+    """
+
+    def test_basic(self):
+        # results test against R's Barnard's Unconditional Test
+
+        barnard_exact = stats.barnard_exact
+        TX_obs, pvalue = barnard_exact([[7, 12], [8, 3]])
+        assert_almost_equal(
+            [TX_obs, pvalue],
+            [1.894338, 0.068153], decimal=5)
+
+        TX_obs, pvalue = barnard_exact(
+            [[43, 40], [10, 39]])
+        assert_almost_equal(
+            [TX_obs, pvalue],
+            [-3.55541, 0.000362832], decimal=5)
+
+        res = barnard_exact([[100, 2], [1000, 5]])[1]
+        assert_approx_equal(res, 0.13507, significant=2)
+        res = barnard_exact([[5, 1], [10, 10]])[1]
+        assert_approx_equal(res, 0.156278, significant=6)
+        res = barnard_exact([[5, 15], [20, 20]])[1]
+        assert_approx_equal(res, 0.0663635, significant=6)
+        res = barnard_exact([[5, 16], [20, 25]])[1]
+        assert_approx_equal(res, 0.116984, significant=6)
+        res = barnard_exact([[10, 5], [10, 1]])[1]
+        assert_approx_equal(res, 0.177537, significant=6)
+        res = barnard_exact([[5, 0], [1, 4]])[1]
+        assert_approx_equal(res, 0.0136719, significant=5)
+        res = barnard_exact([[0, 1], [3, 2]])[1]
+        assert_approx_equal(res, 0.509668, significant=5)
+        res = barnard_exact([[0, 2], [6, 4]])[1]
+        assert_approx_equal(res, 0.197019, significant=5)
+        res = barnard_exact([[2, 7], [8, 2]])
+        assert_approx_equal(res[1], 0.0192108, significant=5)
+        assert_approx_equal(res[0], 2.51847, significant=5)
```

Comment:
Values used form R should be computed with the finest `dp`.

Reply:
I think I am too precise, whatever the dp. When dp gets nearer 100 000, either it is to slow, either it is killed by my computer